apps/web-client/src/lib/db/cr-sqlite/books.ts

```diff
@@ -1,5 +1,47 @@
+/**
+ * @fileoverview Book management system
+ *
+ * Book Overview:
+ * - Manages book metadata in the system (ISBN, title, price, etc.)
+ * - Books are the core entities that can be moved between warehouses
+ * - Book records are referenced by transactions in notes
+ * - Book data includes:
+ *   - ISBN (unique identifier)
+ *   - Title
+ *   - Authors
+ *   - Publisher
+ *   - Price
+ *   - Year
+ *   - Editor
+ *   - Out of print status
+ *   - Category
+ *
+ * Data Sources:
+ * - book table: Core book metadata
+ * - book_transaction table: References books in note transactions
+ */
+
 import { type DB, type BookData } from "./types";
 
+/**
+ * Creates a new book record or updates an existing one.
+ * Uses ISBN as the unique identifier for upsert operations.
+ * All fields except ISBN are optional and will only be updated if provided.
+ *
+ * @param {DB} db - Database connection
+ * @param {BookData} book - Book metadata including:
+ *   - isbn: Unique identifier (required)
+ *   - title: Book title
+ *   - authors: Author names
+ *   - publisher: Publishing company
+ *   - price: Book price
+ *   - year: Publication year
+ *   - editedBy: Editor name
+ *   - outOfPrint: Whether book is out of print
+ *   - category: Book category/genre
```

Comment:
I don't think it's useful to list the fields here.
```suggestion
 * @param {BookData} book - Book metadata
```